// muying-rollupvue/dist/muying-rollupvue.es.js
import { openBlock, createElementBlock, createElementVNode, toDisplayString } from 'vue';

var script$1 = {
    name: 'TestDemo1',
    setup() {
        const message = 'hello world';
        return {
            message
        }
    }
};

const _hoisted_1$1 = { class: "wrapper" };
const _hoisted_2$1 = { class: "box1" };

function render$1(_ctx, _cache, $props, $setup, $data, $options) {
  return (openBlock(), createElementBlock("div", _hoisted_1$1, [
    createElementVNode("div", _hoisted_2$1, toDisplayString($setup.message), 1 /* TEXT */)
  ]))
}

function styleInject(css, ref) {
  if ( ref === void 0 ) ref = {};
  var insertAt = ref.insertAt;

  if (!css || typeof document === 'undefined') { return; }

  var head = document.head || document.getElementsByTagName('head')[0];
  var style = document.createElement('style');
  style.type = 'text/css';

  if (insertAt === 'top') {
    if (head.firstChild) {
      head.insertBefore(style, head.firstChild);
    } else {
      head.appendChild(style);
    }
  } else {
    head.appendChild(style);
  }

  if (style.styleSheet) {
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
}

var css_248z$1 = ".wrapper[data-v-4bca44ee] {\n  width: 100px;\n  height: 100px;\n  background: orange;\n}\n.wrapper[data-v-4bca44ee] .box1[data-v-4bca44ee] {\n  background: aquamarine;\n}";
styleInject(css_248z$1);

script$1.render = render$1;
script$1.__scopeId = "data-v-4bca44ee";
script$1.__file = "src/components/TestDemo1/TestDemo1.vue";

function TestDemo1 (Vue) {
  Vue.component(script$1.name, script$1);
}

var script = {
    name: 'TestDemo2',
    setup() {
        const message = 'world hello';
        return {
            message
        }
    }
};

const _hoisted_1 = { class: "wrapper" };
const _hoisted_2 = { class: "box1" };

function render(_ctx, _cache, $props, $setup, $data, $options) {
  return (openBlock(), createElementBlock("div", _hoisted_1, [
    createElementVNode("div", _hoisted_2, toDisplayString($setup.message), 1 /* TEXT */)
  ]))
}

var css_248z = ".wrapper[data-v-34045f4b] {\n  width: 100px;\n  height: 100px;\n  background: orange;\n}\n.wrapper[data-v-34045f4b] .box1[data-v-34045f4b] {\n  background: pink;\n}";
styleInject(css_248z);

script.render = render;
script.__scopeId = "data-v-34045f4b";
script.__file = "src/components/TestDemo2/TestDemo2.vue";

function TestDemo2 (Vue) {
  Vue.component(script.name, script);
}

function index (Vue) {
  Vue.use(TestDemo1);
  Vue.use(TestDemo2);
}

export { index as default };
